ast/edit/match_test.go

```diff
@@ -0,0 +1,419 @@
+					},
+					Value: &ast.IntegerLiteral{
+						Value: 12,
+					},
+				},
+			},
+		},
+		{
+			name: "expression statement",
+			in: &ast.Package{Files: []*ast.File{{
+				Body: []ast.Statement{
+					&ast.ExpressionStatement{
+						Expression: &ast.BinaryExpression{
+							Operator: ast.AdditionOperator,
+							Left:     &ast.IntegerLiteral{Value: 1},
+							Right:    &ast.IntegerLiteral{Value: 2},
+						},
+					},
+					&ast.ExpressionStatement{
+						Expression: &ast.ObjectExpression{
+							Properties: []*ast.Property{
+								{
+									Key: &ast.Identifier{
+										Name: "b",
+									},
+									Value: &ast.IntegerLiteral{
+										Value: 12,
+									},
+								},
+							},
+						},
+					},
+					&ast.ExpressionStatement{
+						Expression: &ast.MemberExpression{
+							Object: &ast.CallExpression{
+								Callee: &ast.MemberExpression{
+									Object:   &ast.Identifier{Name: "a"},
+									Property: &ast.Identifier{Name: "b"},
+								},
+							},
+							Property: &ast.Identifier{Name: "c"},
+						},
+					},
+				},
+			}}},
+			match: &ast.ExpressionStatement{},
+			result: []ast.Node{
+				&ast.ExpressionStatement{
+					Expression: &ast.BinaryExpression{
+						Operator: ast.AdditionOperator,
+						Left:     &ast.IntegerLiteral{Value: 1},
+						Right:    &ast.IntegerLiteral{Value: 2},
+					},
+				},
+				&ast.ExpressionStatement{
+					Expression: &ast.ObjectExpression{
+						Properties: []*ast.Property{
+							{
+								Key: &ast.Identifier{
+									Name: "b",
+								},
+								Value: &ast.IntegerLiteral{
+									Value: 12,
+								},
+							},
+						},
+					},
+				},
+				&ast.ExpressionStatement{
+					Expression: &ast.MemberExpression{
+						Object: &ast.CallExpression{
+							Callee: &ast.MemberExpression{
+								Object:   &ast.Identifier{Name: "a"},
+								Property: &ast.Identifier{Name: "b"},
+							},
+						},
+						Property: &ast.Identifier{Name: "c"},
+					},
+				},
+			},
+		},
+		{
+			name: "file",
+			in: &ast.Package{Files: []*ast.File{{
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID: &ast.Identifier{Name: "f"},
+						Init: &ast.FunctionExpression{
+							Params: []*ast.Property{
+								{Key: &ast.Identifier{Name: "a"}},
+								{Key: &ast.Identifier{Name: "b"}},
+							},
+							Body: &ast.BinaryExpression{
+								Operator: ast.AdditionOperator,
+								Left:     &ast.Identifier{Name: "a"},
+								Right:    &ast.Identifier{Name: "b"},
+							},
+						},
+					},
+					&ast.ExpressionStatement{
+						Expression: &ast.CallExpression{
+							Callee: &ast.Identifier{Name: "f"},
+							Arguments: []ast.Expression{&ast.ObjectExpression{
+								Properties: []*ast.Property{
+									{Key: &ast.Identifier{Name: "a"}, Value: &ast.IntegerLiteral{Value: 2}},
+									{Key: &ast.Identifier{Name: "b"}, Value: &ast.IntegerLiteral{Value: 3}},
+								},
+							}},
+						},
+					},
+				},
+			}}},
+			match: &ast.File{
```

Comment:
I must be missing something, I don't see how `matchFile` matches anything in the body of the file...